### PR TITLE
[LAP 18] Explosion VFX

### PR DESCRIPTION
**Pull request by tuhri444**
TIu9JsaJto

Explosion particle system made using the engine particle system.

### CODE REVIEW

File: applications/sandbox/systems/explosion_particlesystem.hpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+#pragma region Set directions
+            math::vec3 pointPos = emitterPos + vert_position;
+            math::vec3 pointDirection = math::normalize(pointPos - emitterPos);
+#pragma endregion
+#pragma region Set parameter values
+            rendering::particle particularParticle = particleComponent.read();
+            particularParticle.lifeTime = 0;
+            particularParticle.particleVelocity = m_startingVelocity.x * pointDirection * scaleOfEmitter.r;
+            particleComponent.write(particularParticle);
+#pragma endregion
```
Comment (Algo-ryth-mix) [inline applications/sandbox/systems/explosion_particlesystem.hpp]: pragma regions should not replace comments, they are very Visual Studio specific! Also if using them try to fix the indenting

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]

**Review by kukash — APPROVED** [review]

File: applications/sandbox/systems/explosion_particlesystem.hpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    math::color endColor;
+    rendering::model_handle explosionModel;
+
+    float startingLifeTime = 0.0f;
+    bool looping = false;
+    uint spawnRate = 0;
+    uint particleCount = 0;
+    uint maxParticles = 0;
+};
+
```
Comment (Ragingram2) [inline applications/sandbox/systems/explosion_particlesystem.hpp:29]: Don't forget the Doxygen comments

File: legion/engine/rendering/components/particle_emitter.hpp [hunk 3]
```diff
@@ -11,7 +11,15 @@ namespace legion::rendering
     {
         std::vector<ecs::entity_handle> livingParticles;
         std::vector<ecs::entity_handle> deadParticles;
+        bool playAnimation = false;
         ParticleSystemHandle particleSystemHandle;
         bool setupCompleted;
+
```
Comment (Ragingram2) [inline legion/engine/rendering/components/particle_emitter.hpp:17]: Doxygen Comments

**Review by Ragingram2 — CHANGES_REQUESTED** [review]
You're forgetting some doxygen comments

**Review by Algo-ryth-mix — APPROVED** [review]

File: applications/sandbox/systems/explosion_particlesystem.hpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        m_sizeOverLifetime = params.sizeOverLifeTime;
+        m_particleMaterial = params.particleMaterial;
+        m_particleModel = params.particleModel;
+        m_explosionModel = params.explosionModel;
+
+        m_decelerationScalar = params.decelerationScalar;
+        m_endColor = params.endColor;
+        m_beginColor = m_particleMaterial.get_param<math::vec4>("color");
+    }
+
```
Comment (Ragingram2) [inline applications/sandbox/systems/explosion_particlesystem.hpp]: I meant put Doxygen comments on every method in here

File: applications/sandbox/systems/explosion_particlesystem.hpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+            #pragma region Set parameter values
+            //Read particle component to set its lifetime and its velocity.
+            rendering::particle particularParticle = particleComponent.read();
+            particularParticle.lifeTime = 0;
+            particularParticle.particleVelocity = m_startingVelocity.x * pointDirection * scaleOfEmitter.r;
+            particleComponent.write(particularParticle);
+#           pragma endregion
+        }
+    }
+
```
Comment (Ragingram2) [inline applications/sandbox/systems/explosion_particlesystem.hpp:110]: Here too

**Review by Ragingram2 — APPROVED** [review]

**Pull request merged**